src/main/java/service/RacingCarService.java

```diff
@@ -0,0 +1,67 @@
+package service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import domain.Car;
+import view.RacingCarView;
+
+public class RacingCarService {
+    public void startRacing(List<String> carNames, int targetDistance) {
+        // make cars
+        List<Car> cars = makeCars(carNames);
+
+        // move cars
+        RacingCarView.printStartRacing();
+        moveCars(cars, targetDistance);
+
+        // print winner
+        RacingCarView.printResult(getWinnerNames(cars));
+    }
+
+    public String getWinnerNames(List<Car> cars) {
+        int maxDistance = getMaxDistance(cars);
+        List<String> winners = getWinnerCars(cars, maxDistance);
+
+        return String.join(", ", winners);
+    }
+
+    public int getMaxDistance(List<Car> cars) {
+        return cars.stream()
+                .map(Car::getDistance)
+                .max(Integer::compareTo)
+                .get();
+    }
+
+    public List<String> getWinnerCars(List<Car> cars, int maxDistance) {
+        return cars.stream()
+                .filter(car -> car.getDistance() == maxDistance)
```

Comment:
```suggestion
                .filter(car -> car.sameDistance(maxDistance))
```

객체에게 비교하도록 시켜본다면 어떨까요?

참고 : [Tell, don't ask 원칙(TDA 원칙)](https://effectiveprogramming.tistory.com/entry/Tell-dont-ask)

Reply:
Car Class 는 자동차 도메인으로 설계하였습니다.
layered architecture에서  service / domain 의 구조를 나누는 경계점은 '비즈니스 로직'의 여부라고 생각합니다.  
또한 도메인은 비즈니스 로직이 있는 layer고  service는 도메인을 조합하는 layer라고 생각합니다.

distance는 Car 도메인의 속성이라고 생각하지만,
거리가 같은지 비교하는 기능은 비즈니스 로직이라고 생각하지는 않았습니다.

해당 로직은 비즈니스 로직을 포함하고 있는 domain 영역에 포함하기 보다는
도메인 영역에서는 atomic 한 값을 제공하고, 서비스 영역에서 계산하는게 제 생각에는 더 적합해 보였습니다.

서비스 부분에서 해당 부분을 더 추상화 한다면 int를 비교하는 function을 추가하여 체크할 수도 있겠습니다만
간단히 == 문으로 비교해도 큰 문제가 없을듯 하여 이렇게 구현하였습니다.